packages/atlas/src/views/global/NftSaleBottomDrawer/NftForm/NftForm.hooks.ts

```diff
@@ -9,21 +14,56 @@ export const useNftForm = () => {
   const [listingType, setListingType] = useState<Listing>('Auction')
   const [currentStep, setCurrentStep] = useState(0)
 
+  const { convertDurationToBlocks } = useBlockTimeEstimation()
+
   const getTotalDaysAndHoursText = (start: Date, end: Date) => {
     const { days, hours } = intervalToDuration({
       start,
       end,
     })
-    const parsedDays = days ? `${days} ${days > 1 ? 'Days' : 'Day'}` : ''
-    const parsedHours = hours ? `${hours} ${hours > 1 ? 'Hours' : 'Hour'}` : ''
+
+    const parsedDays = days ? pluralizeNoun(days, 'Day') : ''
+    const parsedHours = hours !== undefined ? (hours >= 1 ? pluralizeNoun(hours, 'Hour') : 'Less than an hour') : ''
     return `${parsedDays} ${parsedHours}`
   }
 
   const nextStep = useCallback(() => setCurrentStep((step) => step + 1), [])
   const previousStep = useCallback(() => setCurrentStep((step) => step - 1), [])
 
+  const getNumberOfBlocksAndDaysLeft = (startDate: AuctionDatePickerValue, endDate: AuctionDatePickerValue) => {
+    const isStartDateAndEndDateValid = startDate?.type === 'date' && endDate?.type === 'date'
+    const now = new Date(Date.now())
+
+    if (isStartDateAndEndDateValid) {
+      return {
+        blocks: convertDurationToBlocks(differenceInMilliseconds(endDate.date, startDate.date)),
+        daysAndHoursText: getTotalDaysAndHoursText(startDate.date, endDate.date),
```

Comment:
As a follow up to my point above, I think it's better for the UI to handle text to display, the hook should just return raw data